udata_front/theme/gouvfr/assets/js/components/Form/UploadGroup/UploadModal.vue

```diff
@@ -0,0 +1,228 @@
+                  </button>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </dialog>
+  </Teleport>
+</template>
+<script>
+import { computed, defineComponent, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { templateRef } from '@vueuse/core';
+import Container from '../../Ui/Container/Container.vue';
+import FileCard from '../FileCard/FileCard.vue';
+import InputGroup from '../InputGroup/InputGroup.vue';
+import UploadGroup from './UploadGroup.vue';
+import useFunctionalState from '../../../composables/useFunctionalState';
+import { requiredWithCustomMessage } from '../../../i18n';
+import { useFilesUpload } from '../../../composables/useFilesUpload';
+
+export default defineComponent({
+  components: { Container, FileCard, InputGroup, UploadGroup },
+  emits: ["send"],
+  inheritAttrs: false,
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+    multiple: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props, { emit }) {
+    const { t } = useI18n();
+    const fileModalTitleId = computed(() => props.id + "-title");
+
+    const { allowedExtensions } = useFilesUpload();
+
+    /** @type {import("vue").Ref<HTMLDialogElement | null>} */
+    const modalRef = templateRef('modalRef');
+
+    /** @type {import("vue").Ref<Array<import("../../../types").NewDatasetFile>>} */
+    const modalFiles = ref([]);
+
+    const fileRequired = requiredWithCustomMessage(t("At least one file or link is required."));
+
+    const requiredRules = {
+      modalFiles: { fileRequired },
+    };
+
+    const { getErrorText, getFunctionalState, hasError, reset, validateRequiredRules, v$ } = useFunctionalState({ modalFiles }, requiredRules, requiredRules);
+
+    /**
+     * @type {import("vue").ComputedRef<Record<string, import("../../../types").AccordionFunctionalState>>}
+     */
+    const state = computed(() => {
+      return {
+        modalFiles: getFunctionalState(v$.value.modalFiles.$dirty, v$.value.modalFiles.$error, false),
+      };
+    });
+
+    /**
+     *
+     * @param {string} field
+     */
+    const stateHasError = (field) => hasError(state, field);
+
+    const nonRemoteFiles = computed(() => modalFiles.value.filter(file => file.filetype === "file"));
+
+    /**
+     *
+     * @param {Array<File>} newFiles
+     */
+    const setFiles = (newFiles) => {
+      modalFiles.value = nonRemoteFiles.value;
+      for(const file of newFiles) {
+        modalFiles.value.push({
+          file,
+          description: "",
+          format: guessFormat(file),
+          filesize: file.size,
+          filetype: "file",
+          mime: file.type,
+          title: file.name,
+          type: "main",
+          state: "none"
+        });
+      }
+    }
+
+    /**
+     * Try to guess file format using MIME type or name.
+     * @param {File} file
+     */
+    const guessFormat = (file) => {
```

Comment:
I wonder if we should show an error here if the file is not one of the expected file formats?

Reply:
We can add an error state to the "file card" on step 3 (like the warning for the missing metadata) and show validation errors here. This allows the user to have a visual clue that something is wrong with the file.

Reply:
Done